services/workflows-service/src/collection-flow/collection-flow.service.ts

```diff
@@ -229,6 +239,46 @@ export class CollectionFlowService {
     return flowData;
   }
 
+  private async _persistFileUrlsToDocuments(
+    documents: any[] = [],
+    projectIds: TProjectIds,
+  ): Promise<any> {
+    const fileEntities = await Promise.all(
+      await documents.reduce((filesList, document) => {
```

Comment:
`Promise.all` expects an array of promises, I understand it's probably needed but it's also likely misplaced.

```suggestion
      documents.reduce((filesList, document) => {
```